localstack/services/opensearch/cluster.py

```diff
@@ -461,11 +461,19 @@ def _create_run_command(
         return cmd
 
     def _create_env_vars(self, directories: Directories) -> Dict:
-        return {
+        env_vars = {
             "OPENSEARCH_JAVA_OPTS": os.environ.get("OPENSEARCH_JAVA_OPTS", "-Xms200m -Xmx600m"),
             "OPENSEARCH_TMPDIR": directories.tmp,
         }
 
+        # if the "opensearch-knn" plugin exists and has a "lib" directory, add it to the LD_LIBRARY_PATH
+        # see https://forum.opensearch.org/t/issue-with-opensearch-knn/12633
+        knn_lib_dir = os.path.join(directories.install, "plugins", "opensearch-knn", "lib")
+        if os.path.isdir(knn_lib_dir):
+            env_vars["LD_LIBRARY_PATH"] = f"{knn_lib_dir}:{os.environ.get('LD_LIBRARY_PATH', '')}"
```

Comment:
Is there a reason for this order? This way, knn_lib_dir would override any potential shared libraries provided by LD_LIBRARY_PATH already.
This might make sense, or could lead to (unlikely) problems if someone wants to use LD_LIBRARY_PATH to load some other shared libraries, but knn_lib_dir has one with the same name.

It is very unlikely that this case will arise, I just want to make sure we make a conscious decision in one way or another here.

Reply:
Good catch! I changed the order and ensured that an empty path cannot be added (in case the LD_LIBRARY_PATH is not set on the machine, an empty path would mean the current working directory).